src/realm/obj.cpp

```diff
@@ -783,12 +869,22 @@ void ConstObj::to_json(std::ostream& out, size_t link_depth, std::map<std::strin
         if (ck.get_attrs().test(col_attr_List)) {
             if (type == type_LinkList) {
                 TableRef target_table = get_target_table(ck);
+                auto primary_key_coll = target_table->get_primary_key_column();
                 auto ll = get_linklist(ck);
                 auto sz = ll.size();
 
-                if (!target_table->is_embedded() &&
-                    ((link_depth == 0) || (link_depth == not_found &&
-                                           std::find(followed.begin(), followed.end(), ck) != followed.end()))) {
+                if (output_mode == output_mode_xjson && !target_table->is_embedded() && primary_key_coll) {
```

Comment:
link lists are handled here, it outputs as a list of the primary keys like ["key1", "key2"]. The normal output is to output the table name and the realm object id's so I added this in here